examples/simple-genai-server/main.go

```diff
@@ -0,0 +1,240 @@
+// Copyright 2024 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"net"
+	"net/http"
+	"os"
+	"strconv"
+	"time"
+
+	"agones.dev/agones/pkg/util/signals"
+	sdk "agones.dev/agones/sdks/go"
+)
+
+// Main starts a server that serves as an example of how to integrate GenAI endpoints into your dedicated game server.
+func main() {
+	sigCtx, _ := signals.NewSigKillContext()
+
+	port := flag.String("port", "7654", "The port to listen to traffic on")
+	genAiEndpoint := flag.String("genAiEndpoint", "", "The full base URL to send API requests to to simulate computer (NPC) responses to user input")
+	genAiContext := flag.String("GenAiContext", "", "Context for the GenAI endpoint")
+	prompt := flag.String("Prompt", "", "The first prompt for the GenAI endpoint")
+	simEndpoint := flag.String("simEndpoint", "", "The full base URL to send API requests to to simulate user input")
+	simContext := flag.String("SimContext", "", "Context for the Sim endpoint")
+	numChats := flag.Int("NumChats", 1, "Number of back and forth chats between the sim and genAI")
+
+	var simConn *connection
+	var compConn * connection
+
+	flag.Parse()
+	if ep := os.Getenv("PORT"); ep != "" {
+		port = &ep
+	}
+	if sc := os.Getenv("SimContext"); sc != "" {
+		simContext = &sc
+	}
+	if gac := os.Getenv("GenAiContext"); gac != "" {
+		genAiContext = &gac
+	}
+	if p := os.Getenv("Prompt"); p != "" {
+		prompt = &p
+	}
+	if se := os.Getenv("SimEndpoint"); se != "" {
+		simEndpoint = &se
+		log.Printf("Creating Client at endpoint %s", *simEndpoint)
```

Comment:
I think creating the client should be done outside of checking env variables. Right now if I pass a flag (rather than an environment variable) then the client won't ever be initialized.